DemoApp/Sources/Components/Feedback/CallEndedViewModifier.swift

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import StreamVideo
+import SwiftUI
+
+private struct CallEndedViewModifier<Subview: View>: ViewModifier {
+
+    private final class CallEndedViewModifierState: ObservableObject {
+        @Published var call: Call?
+        @Published var isPresentingSubview: Bool = false
+
+        init() {}
+    }
+
+    private var notificationCenter: NotificationCenter
+    private var subviewProvider: (Call?) -> Subview
+
+    @StateObject private var state: CallEndedViewModifierState = .init()
+
+    init(
+        notificationCenter: NotificationCenter,
+        @ViewBuilder subviewProvider: @escaping (Call?) -> Subview
+    ) {
+        self.notificationCenter = notificationCenter
+        self.subviewProvider = subviewProvider
+    }
+
+    func body(content: Content) -> some View {
+        content
+            .sheet(isPresented: $state.isPresentingSubview) {
+                subviewProvider(state.call)
+            }
+            .onReceive(
+                notificationCenter.publisher(for: .init(CallNotification.callEnded))
```

Comment:
should we be stricter here? Since I believe we are calling `call.leave` even when we cancel a call because of timeout. In those cases I don't think it makes sense to show this screen - only when there was actual call.

Reply:
You are right about this one. I changed the logic and now the modifier observe the activeCall's state. WDYT?